modules/speech-to-text-impl/src/main/java/org/opencastproject/speechtotext/impl/engine/VoskEngine.java

```diff
@@ -0,0 +1,152 @@
+/**
+ * Licensed to The Apereo Foundation under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ *
+ * The Apereo Foundation licenses this file to you under the Educational
+ * Community License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License
+ * at:
+ *
+ *   http://opensource.org/licenses/ecl2.txt
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ */
+
+package org.opencastproject.speechtotext.impl.engine;
+
+import org.opencastproject.speechtotext.api.SpeechToTextEngine;
+import org.opencastproject.speechtotext.api.SpeechToTextEngineException;
+import org.opencastproject.util.ConfigurationException;
+import org.opencastproject.util.IoSupport;
+
+import org.apache.commons.lang3.StringUtils;
+import org.osgi.service.cm.ManagedService;
+import org.osgi.service.component.ComponentContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Modified;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.Dictionary;
+import java.util.List;
+import java.util.Map;
+
+/** Vosk implementation of the Speech-to-text engine interface. */
+@Component(
+    immediate = true,
+    service = {
+        SpeechToTextEngine.class,
+        ManagedService.class
+    },
+    property = {
+        "service.description=Vosk implementation of the SpeechToTextEngine interface",
+        "service.pid=org.opencastproject.speechtotext.impl.engine.VoskEngine"
+    }
+)
+public class VoskEngine implements SpeechToTextEngine, ManagedService {
+
+  private static final Logger log = LoggerFactory.getLogger(VoskEngine.class);
```

Comment:
Not really important, but can we name this `logger` for consistency? That's what basically everywhere else is used.

```suggestion
  private static final Logger logger = LoggerFactory.getLogger(VoskEngine.class);
```